tests/providers/oracle/hooks/test_oracle.py

```diff
@@ -82,6 +102,7 @@ def test_get_conn_sid(self, mock_connect):
     def test_get_conn_service_name(self, mock_connect):
         dsn_service_name = {'dsn': 'ignored', 'service_name': 'service_name'}
         self.connection.extra = json.dumps(dsn_service_name)
+        self.db_hook.__init__()
```

Comment:
seems you should not have to do this?

Reply:
I did because of the way the tests were originally structured.  They create the hook in `setUp` (line 64 in my commit) and then modify the extra in the tests themselves.  Since all the parameter parsing is now happening in the `__init__` it never picks that up.  This seemed to work for picking up those changes without drastically changing the tests.  I considered trying to rework it to function more like the way tests are structured in https://github.com/apache/airflow/blob/main/tests/providers/ssh/hooks/test_ssh.py with a separate connection for each variation.  But I'm not very confident in writing tests in the first place and didn't fully understand how to rewrite all this in that way.

I did feel a little goofy doing it this way but it seemed to work for the purpose just fine and I couldn't think of any negatives aside from it just being goofy.

Reply:
right so the code before, it retrieved the airflow conn object and recreated the client object every time get_conn was called.
and then they changed the airflow conn object resulting in different behavior with client creation time.  and your move of the logic to `__init__` changes this behavior, so that get_conn doesn't re-retrieve the object.
while it's strictly speaking a "breaking" change probably it's unlikely to trip anyone up.  but i think it would be best to adjust the tests so that the code is invoked in a more normal way.  
what you could is apply mock.patch decorator to patch the `get_connection` method (instead of doing it by assignment in setUp) and then set it to return the modified conn object, then instantiate your hook in each test method e.g. `db_hook = OracleHook(...)`.